zfmt/ztests/join.yaml

```diff
@@ -1,17 +0,0 @@
-script: |
```

Comment:
Don't delete this.

Reply:
That's not what I see.
```console
((2a4a2bcca...)) Noahs-Computer:~/brimdata/zed$ git checkout @^ -- zfmt/ztests/join.yaml
((2a4a2bcca...) +) Noahs-Computer:~/brimdata/zed$ make TEST=/zfmt/ztests/join
=== RUN   TestRecordAccessNamed
--- PASS: TestRecordAccessNamed (0.00s)
=== RUN   TestNonRecordDeref
--- PASS: TestNonRecordDeref (0.00s)
=== RUN   TestNormalizeSet
--- PASS: TestNormalizeSet (0.00s)
=== RUN   TestDuplicates
--- PASS: TestDuplicates (0.00s)
=== RUN   TestTranslateNamed
--- PASS: TestTranslateNamed (0.00s)
=== RUN   TestCopyMutateFields
--- PASS: TestCopyMutateFields (0.00s)
=== RUN   TestContextLookupTypeNamedErrors
--- PASS: TestContextLookupTypeNamedErrors (0.00s)
=== RUN   TestContextLookupTypeNamedAndLookupTypeDef
--- PASS: TestContextLookupTypeNamedAndLookupTypeDef (0.00s)
=== RUN   TestContextTranslateTypeNameConflictUnion
--- PASS: TestContextTranslateTypeNameConflictUnion (0.00s)
=== RUN   TestValueValidate
--- PASS: TestValueValidate (0.00s)
=== RUN   TestZed
=== PAUSE TestZed
=== CONT  TestZed
=== RUN   TestZed/zfmt/ztests
=== PAUSE TestZed/zfmt/ztests
=== CONT  TestZed/zfmt/ztests
=== RUN   TestZed/zfmt/ztests/join
=== PAUSE TestZed/zfmt/ztests/join
=== CONT  TestZed/zfmt/ztests/join
    ztest.go:388: zfmt/ztests/join.yaml: expected and actual stdout differ:
        --- expected
        +++ actual
        @@ -1,10 +1,5 @@
         join on x=x p:=a
         ===
        -fork (
        -  =>
        -    reader
        -  =>
        -    reader
        -)
        +reader
         | join on x=x p:=a

--- FAIL: TestZed (0.31s)
    --- FAIL: TestZed/zfmt/ztests (0.00s)
        --- FAIL: TestZed/zfmt/ztests/join (0.37s)
FAIL
FAIL	github.com/brimdata/zed	0.966s
FAIL
make: *** [test-run] Error 1

Reply:
The error isn't actually caught until the kernel pass, which now that I think of it is a mistake. This should be caught in the semantic pass.

Reply:
OK, but removing this leaves us with no test coverage in `zfmt/ztests/` for joins.

Reply:
Alright I'll cede that point to you, but I think the base query should be changed to make it a valid join not one that should be an error on the semantic pass.

Reply:
Yes, making it a valid join definitely make sense.